src/main/java/com/marklogic/spark/DefaultSource.java

```diff
@@ -100,11 +102,37 @@ public boolean supportsExternalMetadata() {
         return true;
     }
 
+    private boolean isReadFilesOperation(Map<String, String> properties) {
+        return properties.containsKey("path") || properties.containsKey("paths");
+    }
+
     private boolean isReadOperation(Map<String, String> properties) {
         return properties.get(Options.READ_OPTIC_QUERY) != null || isReadWithCustomCodeOperation(properties);
     }
 
     private boolean isReadWithCustomCodeOperation(Map<String, String> properties) {
         return Util.hasOption(properties, Options.READ_INVOKE, Options.READ_XQUERY, Options.READ_JAVASCRIPT);
     }
+
+    private StructType inferSchemaFromOpticQuery(Map<String, String> caseSensitiveOptions) {
```

Comment:
No new code here, I just moved this out of the `inferSchema` method to simplify that method.